block-producer/src/batch_builder/batch.rs

```diff
@@ -27,9 +28,10 @@ pub struct TransactionBatch {
     id: BatchId,
     updated_accounts: BTreeMap<AccountId, AccountStates>,
     produced_nullifiers: Vec<Nullifier>,
-    created_notes_smt: BatchNoteTree,
     /// The notes stored `created_notes_smt`
-    created_notes: Vec<NoteEnvelope>,
+    created_notes_smt: BatchNoteTree,
+    created_notes: Vec<NoteCreated>,
```

Comment:
We possibly don't need this field, see comments below.

Reply:
Indeed.